ush/ioda/bufr2ioda/run_bufr2ioda.py

```diff
@@ -0,0 +1,67 @@
+#!/usr/bin/env python3
+import argparse
+import os
+from gen_bufr2ioda_json import gen_bufr_json
+from wxflow import Logger, Executable, cast_as_dtype
+
+# Initialize root logger
+logger = Logger('run_bufr2ioda.py', level='INFO', colored_log=True)
+
+
+def bufr2ioda(CDATE, RUN, DMPDIR, config_template_dir, COM_OBS):
+    logger.info(f"Process {CDATE} {RUN} from {DMPDIR} to {COM_OBS} using {config_template_dir}")
+
+    # Derived parameters
+    PDY = CDATE[:8]
+    cyc = CDATE[8:10]
+
+    # Load configuration
+    config = {
+        'RUN': RUN,
+        'PDY': PDY,
+        'cyc': cyc,
+        'DMPDIR': DMPDIR,
+        'COM_OBS': COM_OBS
+    }
+    config['current_cycle'] = cast_as_dtype(CDATE)
+
+    # Get gdasapp root directory
+    DIR_ROOT = os.path.realpath(os.path.join(os.path.dirname(os.path.realpath(__file__)), "../../.."))
+    USH_IODA = os.path.join(DIR_ROOT, "ush", "ioda", "bufr2ioda")
+
+    # Create output directory if it doesn't exist
+    os.makedirs(COM_OBS, exist_ok=True)
+
+    # Specify observation types to be processed by a script
+    BUFR_py = ["satwind_amv_goes"]
+
+    for obtype in BUFR_py:
+        logger.info(f"Convert {obtype}...")
+        json_output_file = os.path.join(COM_OBS, f"{obtype}_{PDY}{cyc}.json")
+        template = config_template_dir + '/bufr2ioda_' + obtype + '.json'
+        gen_bufr_json(config, template, json_output_file)
+
+        # Use the converter script for the ob type
+        bufr2iodapy = USH_IODA + '/bufr2ioda_' + obtype + ".py"
+        cmd = Executable(bufr2iodapy)
+        cmd.add_default_arg('-c')
+        cmd.add_default_arg(json_output_file)
+        logger.info(f"Executing {cmd}")
+        cmd()
+
+        # Check if the converter was successful
+        if os.path.exists(json_output_file):
+            os.remove(json_output_file)
+        else:
+            print(f"Problem running converter script for {obtype}")
+
+
+if __name__ == "__main__":
+    parser = argparse.ArgumentParser(description='Convert bufr dump file to ioda format')
+    parser.add_argument('CDATE', type=str, help='YYYYMMDDHH date to process')
```

Comment:
```suggestion
    parser.add_argument('current_cycle', help='current cycle to process',
                            required=True, type=lambda dd: to_datetime(dd))
```
There is no need to use upper case for things like `CDATE`.  That is a convention used in shell scripts to define (environment) variables.
I would also suggest standardizing against `current_cycle`

Reply:
Suggestion accepted at cff2ce5  with one exception.   Inclusion of `required=True` generated the run time error

`TypeError: 'required' is an invalid argument for positionals`

Remove `required=True` and script executes.  

Added `to_datetime` to `from wxflow import` line at top of `run_bufr2ioda.py`.